Correct CPU freq detection on constant-speed systems

randomdude opened the pull request:
On systems where clockspeed is constant, psutils will return zero for max speed. I'm seeing this on my VMWare ESXi guests. This patch will use the current clockspeed if no max is available.

cooperdk commented:
randomdude,

please get a hold of me (info@ppmedia.dk or via discord)...
I tried to merge your pull requests, but most of them caused various errors.

For example, there was a specific new function using Django (in scrapers.scanners) calling a .find on Websites.object which I suspect should have been .filter as .find doesn't exist.

Various other errors were introduced, which caused me to revert and manually introducing some of your code changes, like the case-insensitive file extension and your cpu frequency detection.

Please, write on Discord (preferred) :-)

randomdude commented:
Ah, sorry they caused issues! I changed a load of stuff for my own fork - mostly the logging, and trying to refactor the scrapers - but the changes were pretty big so I didn't try merging them or opening a PR. I thought the changes I did submit here were small and self-contained enough they'd just work okay - sorry!

I'll get Discord set up and we can chat more.

FYI I haven't worked with Django before, so I suspect you're right that I introduced a silly error like that. :$

cooperdk commented:
It's quite ok.

I made the mistake of applying it to my own test setup. I think what we need to do is split the changes up in smaller bits if possible, fx introducing a thing or two at a time and check it out. I like lots of your changes and I have applied a few of them (with credit).

Django is a bit special like most framework, but it is a nice one.
I want to move from AngularJS to Angular for the client/UI but that is "nice to have".

Also, want to enable scene jump by thumbnail if you know what I mean.

I have been working a bit with the scrapers and I have now moved the scanners (clients) to their own area. (videos.scanners.) I am also working with a bit of refactoring.

There is already a logfile in Yapo, which I guess could be used to print out to a log page. I would suggest that because 1) on Linux, it could be handled by the system, and 2) the logger will make the database operations slower due to the way sqlite works (well probably not much though).
I am however thinking about moving certain settings to the database.

Will be looking forward to your msg on Discord... this is the YAPO server: https://discord.gg/zdm7Mdg
